src/inausoft.netCLI.Tests/RegexArgumentDeserializerTests.cs

```diff
@@ -46,52 +52,91 @@ public void RegexArgumentHandler_DeserializeProperCommand_ForStringOptions(strin
                                 "--stringOption", stringOptionValue
                             };
 
-            var deserializer = new RegexOptionsDeserializer();
-
             //Act
-            var command = deserializer.Deserialize<Command1>(args);
+            var command = Deserializer.Deserialize<Command2>(args);
 
             //Assert
             Assert.AreEqual(stringOptionValue, command.StringOption);
         }
 
-        [ExpectedException(typeof(InvalidOptionException))]
         [TestMethod]
-        public void RegexArgumentHandler_ThrowsInvalidOptionException_ForInvalidOption()
+        public void RegexArgumentHandler_DeserializeProperCommand_WhenOptionalCommandWasNotSupplied()
+        {
+            //Arrange
+            var args = new string[] { };
+
+            //Act
+            var command = Deserializer.Deserialize<Command2>(args);
+
+            //Assert
+            Assert.AreEqual(Command2.SomeDefaultValue, command.StringOption);
+        }
+
+        [TestMethod]
+        public void RegexArgumentHandler_ThrowsDeserializationException_ForInvalidOption()
         {
             //Arrange
             var args = new string[]
                             {
+                                "--boolOption", "true",
                                 "--boolOptionXX", "true",
+                                "--stringOption", "stringOptionValue",
+                                "--intOption", "1"
                             };
 
-            var deserializer = new RegexOptionsDeserializer();
+            //Act
+            var exception = Assert.ThrowsException<CommandDeserializationException>(() => Deserializer.Deserialize<Command1>(args));
+
+            //Assert
+            Assert.AreEqual(ErrorCode.UnrecognizedOption, exception.ErrorCode);
+        }
+
+        [TestMethod]
```

Comment:
ExpectedException property?

Reply:
Not sure what do you mean here.. - elaborate please